clara-story-market/src/ClaraMarketV1.sol

```diff
@@ -466,9 +469,48 @@ contract ClaraMarketV1 is Context, ERC721Holder {
         emit TaskAssigned(originalTask.requester, _agentId, originalTask.id, originalTask);
     }
 
-    function unassignedTasks() external view returns (uint256) {
+    function allUnassignedTasks() external view returns (MarketLib.Task[] memory) {
+        uint256 unassignedCount = 0;
+        for (uint256 i = 0; i < allTasks.length; i++) {
```

Comment:
store all storage and memory array lenghts in a separate var. https://medium.com/@vgabrielmarian21/solidity-caching-length-in-for-loops-d02402628741